kernel/printk/printk.c

```diff
@@ -353,28 +357,53 @@ DEFINE_RAW_SPINLOCK(logbuf_lock);
 
 #ifdef CONFIG_PRINTK
 DECLARE_WAIT_QUEUE_HEAD(log_wait);
-/* the next printk record to read by syslog(READ) or /proc/kmsg */
-static u64 syslog_seq;
-static u32 syslog_idx;
-static enum log_flags syslog_prev;
-static size_t syslog_partial;
-
-/* index and sequence number of the first record stored in the buffer */
-static u64 log_first_seq;
-static u32 log_first_idx;
-
-/* index and sequence number of the next record to store in the buffer */
-static u64 log_next_seq;
-static u32 log_next_idx;
 
-/* the next printk record to write to the console */
-static u64 console_seq;
-static u32 console_idx;
-static enum log_flags console_prev;
-
-/* the next printk record to read after the last 'clear' command */
-static u64 clear_seq;
-static u32 clear_idx;
+/*
+ * This control block collects tracking offsets for the log and also, when
+ * enabled via CONFIG_LOGBUFFER, allows log sharing between the bootloader
+ * and the kernel.
+ * NOTE:
+ *   By convention, the control block and the log buffer are contiguous.
+ *   This requirement can be relaxed, if desired.
+ */
+typedef struct {
+
+        /* Pointer to log buffer space and length of space */
+        char *log_buf;
+        u32 log_length;
```

Comment:
This looks to be a replacement for `log_buf_len` throughout the remainder of the code.  If that's the case, why is it being renamed?  Historical accident?  Does it actually behave differently or indicate something different?